roles/gluster_hci/tasks/prerequisites.yml

```diff
@@ -55,7 +55,26 @@
   register: disk_size
   failed_when: disk_size.stdout|int < gluster_features_min_disk|default(15360)
   when: gluster_features_force_diskcheck | default(true)
-  tags: diskcheck
+
+# We reference variables from gluster-infra to check if the disks have
+# logical block size of 512 bytes. To disable the check set
+# gluster_features_512B_check to false. DELETE the below two tasks once
+# OVirt limitation is fixed
+- name: Check if disks have logical block size of 512B else fail
+  command: cat /sys/block/{{item.pvname|basename}}/queue/logical_block_size
+  register: logical_blk_size
+  failed_when: logical_blk_size.stdout|int != 512
+  when: gluster_infra_volume_groups is defined and
+        gluster_features_512B_check|default(true)
+  with_items: "{{ gluster_infra_volume_groups }}"
+
+- name: Check if disks have logical block size (VDO disks) of 512B else fail
+  command: cat /sys/block/{{item.device|basename}}/queue/logical_block_size
+  register: logical_blk_size
+  failed_when: logical_blk_size.stdout|int != 512
+  when: gluster_infra_vdo is defined and
+        gluster_features_512B_check|default(true)
+  with_items: "{{ gluster_infra_vdo }}"
```

Comment:
if vdo is used on the disks, is the disk check skipped? that's what we would want

Reply:
For both vdo and non-vdo the logical block size is checked for the underlying disks. Do you mean that if we are using vdo we shouldn't be checking logical block size?